REGRESSION(253037@main): [ iOS macOS ] http/tests/workers/service/basic-timeout.https.html is a constant timeout

cdumez opened the pull request:
#### a7d733519a166f912e7a6aa80d8662aeb30fb352
<pre>
REGRESSION(253037@main): [ iOS macOS ] http/tests/workers/service/basic-timeout.https.html is a constant timeout
<a href="https://bugs.webkit.org/show_bug.cgi?id=244001">https://bugs.webkit.org/show_bug.cgi?id=244001</a>
&lt;rdar://98739421&gt;

Reviewed by Youenn Fablet.

After 253037@main, we no longer terminate service workers that do not respond in a timely
fashion to loads, unless those loads are navigations.

This test was testing the hung service worker termination logic using fetch requests
and would thus no longer behave as expected after 253037@main. To address the issue,
I rewrote the test to use iframe navigations instead of fetch requests.

* LayoutTests/http/tests/workers/service/basic-timeout.https-expected.txt:
* LayoutTests/http/tests/workers/service/basic-timeout.https.html:
* LayoutTests/http/tests/workers/service/resources/basic-timeout.js:
(async test.addTestFrame):
(async test.frameText):
(async test.testFrame4Loaded):
(async test.frameLoaded):
(async test):
(async test.finishThisTest): Deleted.
(async test.try): Deleted.
(async test.try.checkSuccessAgain): Deleted.
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/253540@main">https://commits.webkit.org/253540@main</a>
</pre>

youennf commented on LayoutTests/http/tests/workers/service/resources/basic-timeout.js:
```diff
@@ -1,75 +1,79 @@
-        finishSWTest();
+        let iframe = frame.contentDocument.createElement("iframe");
+        iframe.src = url;
+        let iframeTimeoutHandle = setInterval(() => {
```
iframe and iframeTimeoutHandle could be const

youennf commented on LayoutTests/http/tests/workers/service/resources/basic-timeout.js:
```diff
@@ -1,75 +1,79 @@
-        }, function(error) {
-            log(error);
-        });
+    frameText = function(testFrame)
```
Why frameText = function(testFrame) instead of function frameText(testFrame)?

cdumez replied:
It needs to have global scope otherwise I couldn't use it inside shouldBeEqualToString().

youennf commented on LayoutTests/http/tests/workers/service/resources/basic-timeout.js:
```diff
@@ -1,75 +1,79 @@
+    frameText = function(testFrame)
+    {
+        let innerText = testFrame.contentDocument.body.innerText;
+        let newLineIndex = innerText.indexOf('\n');
```
could be const

cdumez committed:
REGRESSION(253037@main): [ iOS macOS ] http/tests/workers/service/basic-timeout.https.html is a constant timeout
https://bugs.webkit.org/show_bug.cgi?id=244001
<rdar://98739421>

Reviewed by Youenn Fablet.

After 253037@main, we no longer terminate service workers that do not respond in a timely
fashion to loads, unless those loads are navigations.

This test was testing the hung service worker termination logic using fetch requests
and would thus no longer behave as expected after 253037@main. To address the issue,
I rewrote the test to use iframe navigations instead of fetch requests.

* LayoutTests/http/tests/workers/service/basic-timeout.https-expected.txt:
* LayoutTests/http/tests/workers/service/basic-timeout.https.html:
* LayoutTests/http/tests/workers/service/resources/basic-timeout.js:
(async test.addTestFrame):
(async test.frameText):
(async test.testFrame4Loaded):
(async test.frameLoaded):
(async test):
(async test.finishThisTest): Deleted.
(async test.try): Deleted.
(async test.try.checkSuccessAgain): Deleted.
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:

Canonical link: https://commits.webkit.org/253540@main
